webui/react/src/components/NotebookModal.tsx

```diff
@@ -0,0 +1,231 @@
+import { Button, InputNumber, Modal } from 'antd';
+import { Form, Input, Select } from 'antd';
+import { ModalProps } from 'antd/es/modal/Modal';
+import yaml from 'js-yaml';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import useStorage from 'hooks/useStorage';
+import { getResourcePools, getTaskTemplates } from 'services/api';
+import { NotebookConfig, RawJson, ResourcePool, ResourceType, Template } from 'types';
+import { launchNotebook, previewNotebook } from 'utils/task';
+
+import Link from './Link';
+import css from './NotebookModal.module.scss';
+import RadioGroup from './RadioGroup';
+import Spinner from './Spinner';
+
+const MonacoEditor = React.lazy(() => import('react-monaco-editor'));
+
+const { Option } = Select;
+const { Item } = Form;
+
+const STORAGE_PATH = 'notebook-launch';
+const STORAGE_KEY = 'notebook-config';
+
+interface Props extends ModalProps {
+  onLaunch?: () => void;
+  visible?: boolean;
+}
+
+const NotebookModal: React.FC<Props> = (
+  { visible = false, onLaunch, ...props }: Props,
+) => {
+  const storage = useStorage(STORAGE_PATH);
+  const [ showFullConfig, setShowFullConfig ] = useState(false);
+  const [ templates, setTemplates ] = useState<Template[]>([]);
+  const [ resourcePools, setResourcePools ] = useState<ResourcePool[]>([]);
+  const [ showResourceType, setShowResourceType ] = useState(true);
```

Comment:
**chore:** this should be `false` to begin with and once the conditions are satisfied below, it can be set to `true`.

![2021-05-24 at 8 17 AM](https://user-images.githubusercontent.com/220971/119365354-141f8800-bc6d-11eb-8e41-4c3793df3389.png)